packages/cdktf-cli/lib/dependencies/dependency-manager.ts

```diff
@@ -249,10 +242,11 @@ export class DependencyManager {
    */
   private convertPackageName(name: string): string {
     switch (this.targetLanguage) {
-      case Language.GO:
-        throw Errors.Internal(
-          "pre-built providers are not supported for Go yet"
-        );
+      case Language.GO: // e.g. github.com/hashicorp/cdktf-provider-opentelekomcloud-go
+        return `github.com/hashicorp/cdktf-provider-${name.replace(
+          "@cdktf/provider-",
+          ""
```

Comment:
package name requires a `/${providerName}` at the end. See [here](https://github.com/hashicorp/terraform-cdk/issues/723#issuecomment-1189473189).
So, maybe pull the name into a variable and use it twice?